pypit/arcyarc.pyx

```diff
@@ -438,6 +438,7 @@ def detections_sigma(np.ndarray[DTYPE_t, ndim=1] pixels not None,
             d=1
             flag = 0
             while True:
+                if p+d >= (sz_p-1): break
```

Comment:
I'm wrong - the way you have it is correct on line 441, but I think line 452 should be `if p+d >= sz_p: break`